fvm/state/accounts.go

```diff
@@ -297,11 +329,109 @@ func (a *Accounts) SetCode(address flow.Address, code []byte) error {
 		return nil
 	}
 
-	a.ledger.Set(string(address.Bytes()), string(address.Bytes()), keyCode, code)
+	a.ledger.Set(string(address.Bytes()), string(address.Bytes()), a.contractKey(name), code)
 
 	return nil
 }
 
 func newLedgerGetError(key string, address flow.Address, err error) error {
 	return fmt.Errorf("failed to read key %s on account %s: %w", key, address, err)
 }
+
+func (a *Accounts) setContracts(contracts ContractsList, address flow.Address) error {
+	ok, err := a.Exists(address)
+	if err != nil {
+		return err
+	}
+
+	if !ok {
+		return fmt.Errorf("account with address %s does not exist", address)
+	}
+
+	newContracts, err := rlp.NewEncoder().Encode(contracts)
```

Comment:
Do you think we should store each contract in its own register slot?

RLP was a solution we used for account keys before we had given sufficient thought to how storage should work. I don't think we have to use it here unless there's a compelling need

Reply:
The code of each contract is stored in a separate register – this "contracts" value / RLP list is the set of all contract names. 

@janezpodhostnik I was also confused about the naming. Some ideas to alleviate that:

- Maybe rename `ContractList` to `ContractNames`
- Maybe rename `Accounts.GetContracts` to `GetContractNames` (https://github.com/onflow/flow-go/pull/7/files#diff-177914ac132c7b58a325a8cd8eda6faf6d515c16c11a95ca99fe24a0e2dfff64R384). `GetContract` returns the contract code, but the current plural (`GetContracts`) only returns the names.
- Rename the variables in the function similarly from "contracts" to "contractNames", e.g. `newContracts` to `newContractNames`, `prevContracts` to `prevContractNames`, etc.

Reply:
Renaming was a good idea to make it clearer whats actually stored. I also switched to cbor encoder/decoder, or did you have something else in mind @psiemens?

Reply:
@janezpodhostnik Renaming looks good! 

I don't know what's commonly used for serialization on the EN/FVM side, might be good to keep it consistent.

Reply:
Ah, I see! I was a bit confused before.

Yep, CBOR is a good choice.

I suppose it would be even better if we could do a prefix scan for all of the `contract_` keys. Is that why you brought it up on Slack, @turbolent? It looks like @ramtinms and @m4ksio are planning on adding support for that. How hard do you think it would be to remove this list in the future?

Reply:
The hard part about switching this to prefixes would be the register migration. The code change would be very minor.

Reply:
@psiemens I had asked about the prefix scan on Slack because we need it for general iteration over all storage paths of an account, but yeah, it could be used here as well.